bgpd/bgp_ovsdb_if.c

```diff
@@ -2663,45 +2663,38 @@ bgp_check_neighbor_clear_soft_in (struct ovsdb_idl *idl,
     if (clear_bgp_neighbor_table_requested >
         clear_bgp_neighbor_table_performed) {
         if (object_is_peer_group(neighbor_row)) {
-            daemon_bgp_clear_request(NULL, AFI_IP6, SAFI_UNICAST, clear_group,
+            daemon_bgp_clear_request(NULL, AFI_IP, SAFI_UNICAST, clear_group,
                                      BGP_CLEAR_SOFT_IN, name);
         } else {
-            daemon_bgp_clear_request(NULL, AFI_IP6, SAFI_UNICAST, clear_peer,
+	    afi = network2afi(name);
+            daemon_bgp_clear_request(NULL, afi, SAFI_UNICAST, clear_peer,
                                      BGP_CLEAR_SOFT_IN, name);
         }
 
         clear_bgp_neighbor_table_performed++;
-        clear_bgp_neighbor_table_requested = clear_bgp_neighbor_table_performed;
         snprintf(clear_bgp_neighbor_table_str_performed, MAX_BUF_LEN-1, "%d",
                  clear_bgp_neighbor_table_performed);
-        snprintf(clear_bgp_neighbor_table_str_requested, MAX_BUF_LEN-1, "%d",
-                 clear_bgp_neighbor_table_requested);
 
         ovs_neighbor = get_bgp_neighbor_context(idl, name);
         if (ovs_neighbor) {
-            VLOG_INFO("Adding smap\n");
-            smap_init(&smap_status);
-            smap_add(&smap_status,
+
```

Comment:
We shouldn't reset session state. We reuse old state of smap and increase performed counters for it. Old functionality will be reused for "clear hard"  realisation.